src/Network.cpp

```diff
@@ -1000,81 +991,88 @@ void Network::show_heatmap(const FastState* const state,
     }
 }
 
-void Network::fill_input_plane_pair(const FullBoard& board,
-                                    BoardPlane& black, BoardPlane& white) {
-    auto idx = 0;
-    for (auto j = 0; j < BOARD_SIZE; j++) {
-        for(auto i = 0; i < BOARD_SIZE; i++) {
-            const auto vtx = board.get_vertex(i, j);
-            const auto color = board.get_square(vtx);
-            if (color != FastBoard::EMPTY) {
-                if (color == FastBoard::BLACK) {
-                    black[idx] = true;
-                } else {
-                    white[idx] = true;
-                }
-            }
-            idx++;
+void Network::gather_features(const GameState* const state, NNPlanes& planes) {
+    
+    std::vector<net_t>features_vector;
+    gather_features_vector(state, features_vector, 0);
```

Comment:
If you're going to do it this way, it might as well return the vector. (And this function might, too)

Reply:
So should the `BoardPlane` and `NNPlanes` types be moved to Training.cpp for use when streaming and dumping the training data or are you thinking that it would be simpler to remove the use of `bitset`s and to iterate through a vector in those cases instead?

Reply:
I haven't done anything about this for now. Would you mind if I did further tidy up to Training.cpp in a separate follow-up PR?